MagazineLayout/LayoutCore/ModelState.swift

```diff
@@ -401,31 +402,79 @@ final class ModelState {
         toPreferredHeight: preferredHeight,
         forItemAt: currentIndexPath,
         sectionModels: &currentSectionModels)
+      invalidateSectionMaxYsCacheForSectionIndices(startingAt: currentIndexPath.section)
     }
   }
 
-  func setHeader(_ headerModel: HeaderModel, forSectionAtIndex sectionIndex: Int) {
-    currentSectionModels[sectionIndex].setHeader(headerModel)
+  func updateHeaderHeight(
+    toPreferredHeight preferredHeight: CGFloat,
+    forSectionAtIndex sectionIndex: Int)
+  {
+    func updateHeaderHeight(
+      toPreferredHeight preferredHeight: CGFloat,
+      forSectionAtIndex sectionIndex: Int,
+      sectionModels: inout [SectionModel])
+    {
+      guard sectionIndex < sectionModels.count else {
+        assertionFailure("Updating the preferred height for a header model at section index \(sectionIndex) is out of bounds")
+        return
+      }
 
-    invalidateSectionMaxYsCacheForSectionIndices(startingAt: sectionIndex)
```

Comment:
Same thing here - we only want to invalidate the section max Ys cache for index paths that are valid in the post-batch-update world, since the cache is only for the "current" state.